contrib/win32/win32compat/ssh-agent/keyagent-request.c

```diff
@@ -400,11 +402,11 @@ process_sign_request(struct sshbuf* request, struct sshbuf* response, struct age
 				if (provider)
 					free(provider);
 				if (pin) {
-					SecureZeroMemory(pin, sizeof(pin));
+					SecureZeroMemory(pin, (DWORD)pin_len);
 					free(pin);
 				}
 				if (epin) {
-					SecureZeroMemory(epin, sizeof(epin));
+					SecureZeroMemory(epin, (DWORD)epin_len);
```

Comment:
Aren't we mixing epin_len from iteration n + 1 with epin from iteration n?

Reply:
Yup. good catch! I've moved the section where we free up provider/pin/epin to the end of each iteration where we know the length of each of the strings and are done using them for the current iteration.
I've tested this with two smartcards loaded into the agent, each with a different length pin, and a debugger attached, to make sure everything is aligned. but please double-check my logic. Thanks!

Reply:
Thank you! FWIW, the new code looks good to me.